libkbfs/md_journal.go

```diff
@@ -196,13 +196,15 @@ func makeMDJournalWithIDJournal(
 	}
 
 	if earliest != nil {
-		if earliest.BID() != latest.BID() {
+		if earliest.BID() != latest.BID() &&
+			!(earliest.BID() == NullBranchID &&
```

Comment:
Ah, ok. Can you add a comment to this effect to the comment for the  `branchID` field? Something `branchID` is the BranchID that every MD in the journal is set to, except for when it is LocalSquashBranchID, in which case the journal is a bunch of MDs with null branchID followed by a bunch of MDs with bid = LocalSquashBranchID.